src/helpers/runNotificationListener.ts

```diff
@@ -4,28 +4,44 @@ import sendAppleNotification from '@/helpers/sendAppleNotification'
 import sendGoogleNotification from '@/helpers/sendGoogleNotification'
 
 const apnRegex = /^[a-f0-9]{64}$/
+const notificationInterval = 30 * 60 * 1000 // 30 minutes in milliseconds
 
 const rootFeeds = {
   1: 't/startups',
   2: 't/ketlTeam',
 } as { [key: number]: string }
 
 obssContract.on('FeedPostAdded', async (feedId) => {
+  const currentTime = new Date()
   const title = rootFeeds[feedId.toNumber()]
     ? `Someone posted at ${rootFeeds[feedId.toNumber()]}`
     : undefined
+
   const allTokens = await TokenModel.find()
-  allTokens.forEach(async ({ token }) => {
+
+  for (const tokenDoc of allTokens) {
     try {
-      // APN token
-      if (apnRegex.test(token)) {
-        await sendAppleNotification(token, title)
-      } else {
-        // FCM token
-        await sendGoogleNotification(token, title)
+      const { token, lastSentTime } = tokenDoc
+
+      if (
+        title ||
+        !lastSentTime ||
+        currentTime.getTime() - lastSentTime.getTime() >= notificationInterval
+      ) {
+        // APN token
+        if (apnRegex.test(token)) {
+          await sendAppleNotification(token, title)
+        } else {
+          // FCM token
+          await sendGoogleNotification(token, title)
+        }
```

Comment:
Can it be moved into a separate helper function? Pass `token/title` and call the right function inside

Reply:
I don't think we need it